### ISSUE TITLE
what is embedding_routine function used for?  

### ISSUE THREAD

**Issue by Dero19xx** [polimi-ispl/icpr2020dfdc#69]
Hello  @CrohnEngineer,
I had some problems when I ran [train_triplet.py](https://github.com/polimi-ispl/icpr2020dfdc/blob/master/train_triplet.py#L436). I dont unterstand what is embedding_routine function used for? Can you help me to answer this question?

**Comment by CrohnEngineer** [comment]
Hey @Dero19xx ,

The `embedding_routine` function is used during the training loop for computing the embeddings (i.e., the features extracted by the network) for all samples of the training and validation set.  
These embeddings are then plotted using [Tensorboard](https://pytorch.org/docs/stable/tensorboard.html) and in this way we can "see" how the features of REAL and FAKE videos separate as the training converges and the network learns to discriminate between them.
Hope this is clear, but in case just ask whatever question you have in mind!
Bests
